concepts/enums/about.md

```diff
@@ -1 +1,136 @@
+    #[derive(Drop, Serde, Copy, starknet::Store)]
+    struct Move {
+        up: u32,
+        down: u32,
+        right: u32,
+        left: u32,
+    }
+
+    #[derive(Drop, Serde, Copy, starknet::Store)]
+    enum UserCommand {
+        Login,
+        UpdateProfile,
+        Logout,
+    }
+
+    #[derive(Drop, Serde, Copy, starknet::Store)]
+    enum Action {
+        Quit,
+        Direction: Move,
+        SendMessage: felt252,
+        ChangeAvatarColor: (u8, u8, u8),
+        ProfileState: UserCommand
+    }
+```
+
+An enum can be declared with a combination of custom data types as its variants. In the example above, the `Action` is the complex enum that as a collection of `Quit` (variant with no associated value), `Direction` (variant with the `Move` struct associated value), `SendMessage` (a single felt252 variant), `ChangeAvatar` (variant with the tuple of three associated value), and `ProfileState` (variant with an enum associated value).
+
+You can declare an enum inside and outside of a contract. If declared outside, then it should be imported inside a contract with the `use` keyword.
+
+## Trait Implementations for Enums
+
+In Cairo, traits could be defined and implemented on custom enums. This allows for you to define methods and attributes associated with the enum. We will implement a `Processing` trait on a `Message` enum.
+
+```Rust
+    #[derive(Drop)]
+    enum Message {
+        Quit,
+        Echo: felt252,
+        Move: (u128, u128),
+    }
+```
+
+The `Message` enum is made up of:
+
+~ `Quit` has no associated value.
+
+~ `Echo` is a single felt252 value.
+
+~ `Move` is a tuple of two u128 values.
+
+```Rust
+trait Processing {
+    fn process(self: Message);
+}
+
+impl ProcessingImpl of Processing {
+    fn process(self: Message) {
+        match self {
+            Message::Quit => { println!("quitting") },
+            Message::Echo(value) => { println!("echoing {}", value) },
+            Message::Move((x, y)) => { println!("moving from {} to {}", x, y) },
+        }
+    }
+}
+```
+
+In many situations, enums can come in handy especially when used with the `match` flow as used above in the traits implementation.
+
+Here is how it could be used to process a Quit message:
+
+```Rust
+    let msg: Message = Message::Quit;
+    msg.process();
+```
+
+Running this code would print `quitting`.
+
+## The `Option` Enum and its Advantages
+
+The `Option` enum is a standard Cairo enum that represents the concept of an optional value. It has two variants: `Some: T` and `None`. `Some` variant has an associated value of type `T`, while `None` represents the absence of an associated value.
+
+```Rust
+enum Option<T> {
+    Some: T,
+    None,
+}
+```
+
+The Option enum is helpful because it allows you to explicitly represent the possibility of a value being absent, making your code more expressive and easier to reason about. Using Option can also help prevent bugs caused by using uninitialized or unexpected null values.
```

Comment:
```suggestion
The `Option` enum is helpful because it allows you to explicitly represent the possibility of a value being absent, making your code more expressive and easier to reason about. Using `Option` can also help prevent bugs caused by using uninitialized or unexpected `null` values.
```